src/Aspire.Hosting.Qdrant/QdrantBuilderExtensions.cs

```diff
@@ -0,0 +1,106 @@
+    /// Adds a named volume for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the resource name.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only volume.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataVolume(this IResourceBuilder<QdrantServerResource> builder, string? name = null, bool isReadOnly = false)
+        => builder.WithVolume(name ?? VolumeNameGenerator.CreateVolumeName(builder, "data"), "/var/lib/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithDataBindMount(this IResourceBuilder<QdrantServerResource> builder, string source, bool isReadOnly = false)
+        => builder.WithBindMount(source, "/var/lib/qdrant/storage", isReadOnly);
+
+    /// <summary>
+    /// Adds a bind mount for the init folder to a Qdrant container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> WithInitBindMount(this IResourceBuilder<QdrantServerResource> builder, string source, bool isReadOnly = true)
+        => builder.WithBindMount(source, "/docker-entrypoint-initdb.d", isReadOnly);
+
+    /// <summary>
+    /// Add a reference to a Qdrant settings for a project.
+    /// </summary>
+    /// <param name="builder">An <see cref="IResourceBuilder{T}"/> for <see cref="ProjectResource"/></param>
+    /// <param name="qdrantResource">The Qdrant server resource</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ProjectResource> WithReference(this IResourceBuilder<ProjectResource> builder, IResourceBuilder<QdrantServerResource> qdrantResource)
+    {
+        builder.WithEnvironment(context =>
+        {
+            context.EnvironmentVariables[$"ConnectionStrings__{qdrantResource.Resource.Name}"] = qdrantResource.Resource.ConnectionStringExpression;
+            context.EnvironmentVariables[$"Parameters__{qdrantResource.Resource.ApiKeyParameter.Name}"] = qdrantResource.Resource.ApiKeyParameter.Value;
```

Comment:
Why `Parameters__`? This should be written into a component specific config section. Also you should directly set the parameter, not the parameter value.

e.g.:

```
```suggestion
            context.EnvironmentVariables[$"Qdrant__ApiKey"] = qdrantResource.Resource.ApiKeyParameter;
```

Reply:
Certainly `Qdrant__XXXX` is better.  For component-specific sectio though, since this is named resource value, were you thinking this would actually be `Aspire__Qdrant__[name]__ApiKey`?

Reply:
Actually, looking at other components, is this correct? Looking at MySQL for example, there would be one Aspire:MySqlConnectorSettings:ConnectionString value -- if I have 2 MySQL resources, this would be shared between them as the component config is not per-named resource...correct? /cc @eerhardt